client/src/MoreInfoPage.js

```diff
@@ -0,0 +1,103 @@
+import { BsFillHeartFill } from 'react-icons/bs';
+import { useLocation } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+
+export default function  MoreInfoPage() {
+  return (
+    <MoreInfo/>
+  )
+}
+
+function MoreInfo() {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const readBookObject = location.state;
+
+    async function addBook() {
+    let book = { title: 'Title Unknown', author: 'Author Unknown', isbn: 'Not Found', rating: 0, image: "https://blog.springshare.com/wp-content/uploads/2010/02/nc-md.gif", price: 19.99, quantity: 1 }
+    if (readBookObject.volumeInfo.title) {
+      book.title = readBookObject.volumeInfo.title;
+    }
+    if (readBookObject.volumeInfo.authors) {
+      book.author = readBookObject.volumeInfo.authors[0];
+    }
+    if (readBookObject.volumeInfo.industryIdentifiers[0]) {
+      book.isbn = readBookObject.volumeInfo.industryIdentifiers[0].identifier
+    }
+    if (readBookObject.volumeInfo.averageRating) {
+      book.rating = readBookObject.volumeInfo.averageRating;
+    }
+    if (readBookObject.volumeInfo.imageLinks) {
+      book.image = readBookObject.volumeInfo.imageLinks.thumbnail;
+    }
+    if (readBookObject.saleInfo.retailPrice) {
+      book.price = readBookObject.saleInfo.retailPrice.amount.toFixed(2);
+    }
+    try {
+    const response = await fetch((`/api/cart`), {method: 'POST', headers: {"Content-Type" : "application/json"}, body : JSON.stringify(book)})
+    if (!response.ok) {
+      throw new Error(`Response error: ${response.status}`);
+    }
+    navigate('/checkout');
+
+    }
+    catch (error) {
+      console.log(`There was a post error: ${error.message} `);
+    }
+  }
+
+  async function addToWishlist() {
+    let book = { title: 'Title Unknown', author: 'Author Unknown', isbn: 'Not Found', rating: 0, image: "https://blog.springshare.com/wp-content/uploads/2010/02/nc-md.gif", price: 19.99, description: 'There was no description found for this book.'};
+    if (readBookObject.volumeInfo.title) {
+      book.title = readBookObject.volumeInfo.title;
+    }
+    if (readBookObject.volumeInfo.authors) {
+      book.author = readBookObject.volumeInfo.authors[0];
+    }
+    if (readBookObject.volumeInfo.industryIdentifiers[0]) {
+      book.isbn = readBookObject.volumeInfo.industryIdentifiers[0].identifier;
+    }
+    if (readBookObject.volumeInfo.averageRating) {
+      book.rating = readBookObject.volumeInfo.averageRating;
+    }
+    if (readBookObject.volumeInfo.imageLinks) {
+      book.image = readBookObject.volumeInfo.imageLinks.thumbnail;
+    }
+    if (readBookObject.saleInfo.retailPrice) {
+      book.price = readBookObject.saleInfo.retailPrice.amount.toFixed(2);
+    }
```

Comment:
The above looks very similar to the code in `addBook`. Is it identical? Can you extract some or all of it into a shared function?